refactor(data-table): DataTable components to use aria-label instead of ariaLabel

digiacomo-a opened the pull request:
This is my first contribution here, I would like to learn more.
In the scope of this PR I've marked `ariaLabel` as deprecated (but still usable) and add `aria-label`

No breaking changes, `aria-label` will have priority if provided.

Closes #12802

I think this PR close the issue above with the updates to the DataTable components.

#### Changelog

**New**

- aria-label prop for TableSelectAll
- aria-label prop for TableSelectRow

**Changed**

- ariaLabel is marked as deprecated (console.warn)

**Removed**

- nothing has been removed, no breaking changes (just warning)

#### Testing / Reviewing

I've added a unit-test scenario for all cases to check:
- [x] aria-label is used if provided
- [X] if both aria-label and ariaLabel are provided, we use the aria-label with priority
- [X] if we keep using ariaLabel and not passing aria-label, it will still work but print a warning.

digiacomo-a committed:
refactor(data-table): data-table

Update to deprecate ariaLabel in favor of aria-label.

No breaking changes, aria-label will have priority if provided.

Gururajj77 commented on packages/react/examples/custom-data-table-state-manager-vite/package.json:
Could you please give context on why the versioning of the `eslint-plugin-react-refresh` is changed?

digiacomo-a replied:
I had some issue to run yarn install and commit was failing with an error related to that module.
It was related to something else on my local (I just checkout the repo) but I was keeping this version since it's the latest minor update.

I think the change in that file can be reverted if we want, but it will not have any impact since we use the caret `^` prefix, we are already installing the latest minor update, so it was still installing that version.

Let me know if I'm wrong and if we want to revert it, please :) Thanks for the comment!

Gururajj77 reviewed (approved):
LGTM 🚀

digiacomo-a committed:
Merge branch 'main' into datatable-12802

digiacomo-a committed:
refactor(data-table): data-table

add my user as contributor for code and a11y.

No changes in the logic compared to last commit

digiacomo-a committed:
Merge branch 'main' into datatable-12802

digiacomo-a committed:
Merge branch 'main' into datatable-12802

digiacomo-a committed:
Merge branch 'main' into datatable-12802

digiacomo-a committed:
Merge branch 'main' into datatable-12802

digiacomo-a committed:
Merge branch 'main' into datatable-12802

digiacomo-a committed:
Merge branch 'main' into datatable-12802

digiacomo-a committed:
Merge branch 'main' into datatable-12802

digiacomo-a committed:
Merge branch 'main' into datatable-12802

digiacomo-a committed:
Merge branch 'main' into datatable-12802

digiacomo-a committed:
Merge branch 'main' into datatable-12802

tay1orjones commented on packages/react/src/components/DataTable/TableSelectAll.tsx:
```diff
@@ -10,12 +10,20 @@ import React from 'react';
+  ['aria-label']?: string;
+
+  /**
+   * Deprecated, please use `aria-label` instead.
```
There's actually a formal [deprecated tag in JSDoc](https://jsdoc.app/tags-deprecated) we could use here
```suggestion
   * @deprecated please use `aria-label` instead.
```

digiacomo-a replied:
Sorry, I didn't received emails for this and forgot to monitor the PR, so I missed the suggestions. Thanks for the review and for taking care of the missing parts.

tay1orjones commented on packages/react/src/components/DataTable/TableSelectRow.tsx:
```diff
@@ -11,12 +11,21 @@ import classNames from 'classnames';
+  ['aria-label']?: string;
+
+  /**
+   * Deprecated, please use `aria-label` instead.
```
```suggestion
   * @deprecated please use `aria-label` instead.
```

tay1orjones commented on packages/react/src/components/DataTable/__tests__/DataTable-test.js:
```diff
@@ -296,8 +296,11 @@ describe('DataTable', () => {
+      let spy;
 
       beforeEach(() => {
+        // TODO: Remove the mock of console.warn once we remove ariaLabel from DataTable
```
```suggestion
        // v12 TODO: Remove the mock of console.warn once we remove ariaLabel from DataTable
```

tay1orjones reviewed (approved):
Thanks so much for this! I'll incorporate my suggestions and then merge

tay1orjones committed:
Apply suggestions from code review

tay1orjones committed:
Merge branch 'main' into datatable-12802
